host/tests/dsDisplay_L3_Tests/dsDisplay_test01_GetDisplayEDID.py

```diff
@@ -0,0 +1,181 @@
+        test = self.testSetup.get("assets").get("device").get(self.testName)
+        cmds = test.get("execute")
+        if cmds is not None:
+            for cmd in cmds:
+                self.writeCommands(cmd)
+
+    #TODO: Current version supports only manual verification.
+    def testGetEDID(self, manual=False, port="dsVIDEOPORT_TYPE_HDMI" , index:int = 0):
+        """
+        Gets the EDID of the display.
+        Args:
+            manual (bool, optional): Manual verification (True: manual, False: other verification methods).
+                                    Defaults to other verification methods
+                                    port (str, optional): The type of video port to retrieve the EDID.
+                                    index (int, optional): The index of the port to retrieve the EDID.
+        Returns:
+            bool
+        """
+        # Get the EDID from the display
+        edid = self.testdsDisplay.getEdid(port, index)
+
+        if manual:
+            # Manual verification
+            return self.testUserResponse.getUserYN(f"Is the EDID {edid} correct? (Y/N):")
+        else:
+            # Automation verification
+            #TODO: Add automation verification methods
+            return False
+
+    def testFunction(self):
+        """
+        This function will test the Display by getting the EDID of the display.
+
+        This function:
+        - Downloads the required assets.
+        - Runs the prerequisite commands.
+        - Retrieves EDID for each supported port and verifies them.
+        - Cleans up assets after the test.
+
+        Returns:
+            bool: Final result of the test.
+        """
+
+        # Download the assets listed in the test setup configuration file
+        self.testDownloadAssets()
+
+        # Run Prerequisites listed in the test setup configuration file
+        self.testRunPrerequisites()
+
+        # Create an instance of the dsDisplayClass
+        self.testdsDisplay = dsDisplayClass(self.deviceProfile, self.hal_session)
+
+        # Initialize the dsDisplay module
+        self.testdsDisplay.initialise()
+
+        self.log.testStart(self.testName, '1')
+
+        # Get the supported video ports
+        supported_ports = self.testdsDisplay.getSupportedPorts()
+        result=True
+
+        if not supported_ports:
+            self.log.error("No supported ports found.")
+            return False
+
+        # Loop through the supported video ports
+        for port, index in supported_ports:
+            self.log.info(f"Testing port {port} at index {index}.")
+            edid_result = self.testGetEDID(manual=True, port=port, index=index)
+            if not edid_result:
+                self.log.error(f"EDID verification failed for port {port} at index {index}.")
+                result = False
+                break  # Exit if verification fails
+
+        #Terminate dsDisplay Module
+        self.testdsDisplay.terminate()
```

Comment:
move terminate and del to testEndFunction similar to https://github.com/rdkcentral/rdk-halif-test-device_settings/blob/feature/gh233_dsAudio_L3_python_updates/host/tests/L3_TestCases/dsAudio/dsAudioHelperClass.py#L156